src/adapter/src/rbac.rs

```diff
@@ -156,21 +156,30 @@ pub fn check_plan(
         return Ok(());
     }
 
-    if session.is_superuser() {
-        return Ok(());
-    }
-
     let role_id = session.role_id();
-    let Some(role) = catalog.try_get_role(role_id) else {
+    if catalog.try_get_role(role_id).is_none() {
         // PostgreSQL allows users that have their role dropped to perform some actions,
         // such as `SET ROLE` and certain `SELECT` queries. We haven't implemented
         // `SET ROLE` and feel it's safer to force to user to re-authenticate if their
         // role is dropped.
         return Err(AdapterError::ConcurrentRoleDrop(role_id.clone()));
     };
 
+    if session.is_superuser() {
```

Comment:
Why did this check get moved down?

Reply:
This was somewhat of a judgement call/subjective. I figured that the user originally being authenticated as a superuser is meaningless if someone then went and deleted the role. 

Additionally it helps guarantee that whenever we go to sequence a plan we know the active role will exist in the catalog. It might not be completely necessary but it makes things a bit easier to reason about.